### DISCUSSION TITLE
Key + Mousebinds for "Navigate back" and "Navigate forward"

### DISCUSSION THREAD

**Discussion by ModProg** [spacedriveapp/spacedrive#1014] [Feature Request]
I suggest mapping Mouse4 and Mouse5 as well as “Alt+Left” and “Alt+Right” to “Navigate back” and “Navigate forward” respectively. It might make sense to have “Alt+Up” be “Go to parent directory”.

The suggestion for the default bindings come from web browsers and other file managers using them like this (Firefox and Dolphin at least)

**Comment by brxken128** [comment]
Accessibility is in the plans, and this will be included along with gestures. We can already use alt (cmd on Mac) and arrrow keys to switfch between selected files.

**Reply by ModProg** [reply]
I'd expect just arrow keys to switch selected files, and alt + arrow keys for navigation history.

**Reply by brxken128** [reply]
Well full accessibility will come later, we are in very early alpha and things aren't 100% just yet. We may reesrve arrow keys for other functions, that's undecided, but navigation will be a breeze once things are sorted.

**Comment by utkubakir** [comment]
Good idea, I see myself trying to use those shortcuts a lot as well. Consider them done!

P.S. Arrow keys work just as you mentioned for selected files.

**Comment by bitn00b** [comment]
I don't know if I should open a new Discussion but this is quite similar.

On Mac using Command+Up: "Going to the parent Directory" and Command+Down: "Go into the current selected Directory"  I think those should be also added, just don't know if I should open a new Discussion for that

**Reply by ModProg** [reply]
makes sense, probably best to keep consistency with what the platform otherwise uses, what is the keybind to navigate back in mac?

**Reply by bitn00b** [reply]
```
Command–Left Bracket ([): Go to the previous folder.
Command–Right Bracket (]): Go to the next folder.
Command–Up Arrow: Open the folder that contains the current folder.
Command–Down Arrow: Open the selected item.
```

Reference: https://support.apple.com/en-us/HT201236#finder

Sadly on a German Keyboard triggering the prev / next folder - is nearly impossible (due to how Left / Right Bracket need to be triggered xD) --- so that I might have to use the Keybind Feature to switch it - will there be Keybind Profiles / Templates then these could be shared in the community for a "improved german mac keyboard layout" or something :D

**Reply by utkubakir** [reply]
I think Alt will map to Command on macOS but noted this as well!

**Reply by ModProg** [reply]
I think it makes sense to match what chrome does and support both key binds, i.e., `Command+Left` and `Command+[` (https://support.google.com/chrome/answer/157179?hl=en) Safari has apparently layout dependent key binds for navigating history i.e., `Command+ü` and `Command+ß`. (https://support.apple.com/de-de/guide/safari/cpsh003/mac) "Zurück zur vorherigen Webseite"
